### DISCUSSION TITLE
Questions of future Prestashop v9

### DISCUSSION THREAD

**Discussion by vitalijalbu** [PrestaShop/PrestaShop#34756] [Q&A]
Hi guys, will prestashop 9 have some UI enhancements? Like vuejs or react admjn panel?? Jquery is dead nowdays….
Also somehow will webservices reign over the infrastructure? Headless is the future….

PS. I really appreciate if it will be easier to create entities and relationships from admin panel, like metaobjects of Shopify.

**Comment by matks** [comment]
Hello @vitalijalbu

I think you have a wrong understanding of PrestaShop 😅 it is not a software where you can, in a few months, bring a massive change such as switching back-office technology from PHP-rendered to JS-rendered or change everything to rely on webservices. There are millions of code in PrestaShop, changing them while preserving all the implemented features, making sure not to introduce bugs is a work that will require years, not months 😉 . PrestaShop is a gigantic monolith.

So
> Hi guys, will prestashop 9 have some UI enhancements? Like vuejs or react admjn panel?? Jquery is dead nowdays….

No 😉 

> Also somehow will webservices reign over the infrastructure? Headless is the future….

Probably one day but not in PrestaShop 9.

> PS. I really appreciate if it will be easier to create entities and relationships from admin panel, like metaobjects of Shopify.

Very unlikely as it would bring massive issues

I think you might be interested in reading https://build.prestashop-project.org/news/2019/prestashop-in-2019-and-beyond-part-3-the-future-architecture/

**Comment by kpodemski** [comment]
> Jquery is dead nowdays….

Ha, that's a good one :) PHP is also dead.
